src/components/UserProfile/TeamsAndProjects/UserTeamsTable.jsx

```diff
@@ -10,7 +10,7 @@ import { connect } from 'react-redux';
 const UserTeamsTable = props => {
   const [tooltipOpen, setTooltip] = useState(false);
   const [teamCode, setTeamCode] = useState(props.userProfile? props.userProfile.teamCode: props.teamCode);
-
+  const canToggleInvisibility = props.hasPermission(('toggleInvisibility'));
```

Comment:
```suggestion
  const canToggleInvisibility = props.hasPermission(('toggleInvisibility')) && targetIsEditable;
```
include logic for `canEditVisibility`